### PR TITLE
API-1491: Display error logs in row

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

screenshot-localhost_8080-2021 03 11-17_42_48

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx [hunk 1]
```diff
@@ -22,7 +36,43 @@ type EventSubscriptionLogs = {
     endScrollReached: boolean;
 };
 
-export const EventLogList: FC<{connectionCode: string}> = ({ connectionCode }) => {
+const ExtraSmallColumnHeaderCell = styled(Table.HeaderCell)`
+    width: 9%;
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:40]: Why `%` ? Why not directly a width in px ?

File: src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx [hunk 2]
```diff
@@ -67,17 +117,85 @@ export const EventLogList: FC<{connectionCode: string}> = ({ connectionCode }) =
...
+            case EventSubscriptionLogLevel.WARNING:
+                return 'warning';
+            case EventSubscriptionLogLevel.ERROR:
+                return 'danger';
+            case EventSubscriptionLogLevel.INFO:
+                return 'primary';
+            case EventSubscriptionLogLevel.NOTICE:
+                return 'tertiary';
+        }
+    };
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:131]: tip: You can move this method outside the component, like somewhere upper in the file, the component will be easier to read

Comment (tseho) [reply src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:131]: `getBadgeLevel` ?

**Review by tseho — COMMENTED** [review]
We need to add a least one test for the list.
We can check together, It will be probably linked to the `fetch` PR.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml [hunk 3]
```diff
@@ -232,11 +232,19 @@ akeneo_connectivity.connection:
             link: Learn more about event subscription configuration.
         event_logs:
             event_subscription_disabled:
-                title: The event subscription isn’t enabled for this connection. 
+                title: The event subscription isn’t enabled for this connection.
                 link: Enable the event subscription
             no_event_logs:
                 title: There is no log for the moment.
 
+    event_logs:
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/translations/jsmessages.en_US.yml:240]: I have already created the section `event_logs` and you should probably put this inside a sub-section for the page `event_log_list` (like `event_subscription_disabled` and `no_event_logs`)

File: src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx (hunk 2 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:131]: I will go further and create a custom badge component for that

File: src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx [hunk 4]
```diff
@@ -67,17 +117,85 @@ export const EventLogList: FC<{connectionCode: string}> = ({ connectionCode }) =
...
+                                <ArrowRightIcon/>
+                                <DatetimeContainer>
+                                    <div>
+                                        <StyledDateIcon size={16} color={theme.color.grey100} />
+                                        <Box>
+                                            {new Intl.DateTimeFormat(userLocale, {
+                                                year: 'numeric',
+                                                month: '2-digit',
+                                                day: '2-digit',
+                                            }).format(new Date(timestamp))}
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:172]: To replace with `src/Akeneo/Connectivity/Connection/front/src/shared/formatter/use-date-formatter.ts`, especially as the locale format of the pim is not compatible with the one for `DateTimeFormat` (en_US vs en-US)

**Review by tseho — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/front/tests/src/webhook/components/EventLogBadge.test.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import {EventLogBadge} from '@src/webhook/components/EventLogBadge';
+import {render, screen} from 'akeneo-design-system/lib/storybook/test-util';
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/tests/src/webhook/components/EventLogBadge.test.tsx:4]: You should not depend on `akeneo-design-system` internals.

```suggestion
import {render, screen} from '@testing-library/react';
```
^ not sure about the import

File: src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx [hunk 6]
```diff
@@ -17,16 +58,70 @@ export const EventLogList: FC<{connectionCode: string}> = ({connectionCode}) =>
...
+                                        <StyledClockIcon size={16} color={theme.color.grey100} />
+                                        <Box>
+                                            {dateFormatter(timestamp * 1000, {
+                                                hour: '2-digit',
+                                                minute: '2-digit',
+                                                second: '2-digit',
+                                            })}
+                                        </Box>
+                                    </TimeContainer>
+                                </DatetimeContainer>
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/front/src/webhook/components/EventLogList.tsx:111]: This should probably be in a dedicated component

File: src/Akeneo/Connectivity/Connection/front/tests/src/webhook/components/EventLogBadge.test.tsx (hunk 5 above)
Comment (tseho) [reply src/Akeneo/Connectivity/Connection/front/tests/src/webhook/components/EventLogBadge.test.tsx:4]: It's a failed auto import, I never intended to depends on the dsm for our tests.
Good catch !

**Review by pchasle — APPROVED** [review]

**Comment by tseho** [comment]
I'm merging it like so into the feature branch, the CI failing on `test_deploy` has nothing to do with Willy's PR.

**Pull request merged**